pageserver/src/tenant/storage_layer/layer.rs

```diff
@@ -0,0 +1,1115 @@
+                    &LayerFileMetadata::new(
+                        this.desc.file_size,
+                    ),
+                )
+                .await;
+
+                let result = match result {
+                    Ok(size) => {
+                        timeline.metrics.resident_physical_size_gauge.add(size);
+                        Ok(())
+                    }
+                    Err(e) => {
+                        Err(e)
+                    }
+                };
+
+                if let Err(res) = tx.send(result) {
+                    match res {
+                        Ok(()) => {
+                            // our caller is cancellation safe so this is fine; if someone
+                            // else requests the layer, they'll find it already downloaded
+                            // or redownload.
+                            //
+                            // however, could be that we should consider marking the layer
+                            // for eviction? alas, cannot: because only DownloadedLayer
+                            // will handle that.
+                        },
+                        Err(e) => {
+                            // our caller is cancellation safe, but we might be racing with
+                            // another attempt to initialize. before we have cancellation
+                            // token support: these attempts should converge regardless of
+                            // their completion order.
+                            tracing::error!("layer file download failed, and additionally failed to communicate this to caller: {e:?}");
+                        }
+                    }
+                }
+
+                Ok(())
+            }
+            .in_current_span(),
+        );
+        match rx.await {
+            Ok(Ok(())) => {
+                if let Some(reason) = self
+                    .needs_download()
+                    .await
+                    .map_err(DownloadError::PostStatFailed)?
+                {
+                    // this is really a bug in needs_download or remote timeline client
+                    panic!("post-condition failed: needs_download returned {reason:?}");
+                }
+
+                self.consecutive_failures.store(0, Ordering::Relaxed);
+
+                Ok(())
+            }
+            Ok(Err(e)) => {
+                let consecutive_failures =
+                    self.consecutive_failures.fetch_add(1, Ordering::Relaxed);
+                tracing::error!(consecutive_failures, "layer file download failed: {e:#}");
+                let backoff = utils::backoff::exponential_backoff_duration_seconds(
+                    consecutive_failures.min(u32::MAX as usize) as u32,
+                    1.5,
+                    60.0,
+                );
+                let backoff = std::time::Duration::from_secs_f64(backoff);
+
+                // unless we get cancelled, we will hold off the semaphore init
```

Comment:
the init semaphore. I mean the one heavier_once_cell gives for the one winning initializer.
```suggestion
                // unless we get cancelled, we will hold off the init semaphore
```